sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```
@@ -3425,12 +3428,164 @@ public void run() {
+	 * For sending requests asynchronously, use sendRequests
+	 *
+	 * @param rpcs is the list of RPCRequests being sent
+	 * @param listener listener for updates and completions
+	 * @throws SdlException if an unrecoverable error is encountered
+	 */
+	@SuppressWarnings("unused")
+	public void sendSequentialRequests(final List<RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {
+		if (_proxyDisposed) {
+			throw new SdlException("This object has been disposed, it is no long capable of executing methods.", SdlExceptionCause.SDL_PROXY_DISPOSED);
+		}
+
+		SdlTrace.logProxyEvent("Application called sendSequentialRequests", SDL_LIB_TRACE_KEY);
+
+		synchronized(CONNECTION_REFERENCE_LOCK) {
+			if (!getIsConnected()) {
+				SdlTrace.logProxyEvent("Application attempted to call sendSequentialRequests without a connected transport.", SDL_LIB_TRACE_KEY);
+				throw new SdlException("There is no valid connection to SDL. sendSequentialRequests cannot be called until SDL has been connected.", SdlExceptionCause.SDL_UNAVAILABLE);
+			}
+		}
+
+		if (rpcs == null){
+			//Log error here
+			throw new SdlException("You must send some RPCs", SdlExceptionCause.INVALID_ARGUMENT);
+		}
+
+		int requestCount = rpcs.size();
+
+		// Break out of recursion, we have finished the requests
+		if (requestCount == 0) {
+			listener.onFinished();
+			return;
+		}
+
+		RPCRequest rpc = rpcs.get(0);
+		if (rpc.getCorrelationID() == null) {
+			rpc.setCorrelationID(CorrelationIdGenerator.generateId());
+		}
+
+		rpc.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if (response.getSuccess()) {
+					// success
+					rpcs.remove(0);
+					listener.onUpdate(rpcs.size());
+					try {
+						// recurse after successful response of RPC
+						sendSequentialRequests(rpcs, listener);
+					} catch (SdlException e) {
+						e.printStackTrace();

```

Isn't it better to throw the exception instead of catching it so the developer can handle it?

---

Hmm, my thought was that its not really a library problem at that point, if it got a bad response from core, I don't want it to error out the application. That is different than above where passing in empty or null lists is the developer's problem.